Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -111,18 +274,46 @@ extension PreplannedMapModel {
     }
 }
 
+private extension FileManager {
+    static let packageDirectoryPath: String = "Package"
+    static let mmpkPathExtension: String = "mmpk"
+}
+
+extension PreplannedMapModel: Hashable {
+    nonisolated public static func == (lhs: PreplannedMapModel, rhs: PreplannedMapModel) -> Bool {
+        lhs === rhs
+    }
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(ObjectIdentifier(self))
+    }
+}
+
 /// A type that acts as a preplanned map area.
 protocol PreplannedMapAreaProtocol {
     func retryLoad() async throws
+    func makeParameters(using offlineMapTask: OfflineMapTask) async throws -> DownloadPreplannedOfflineMapParameters?
     
     var packagingStatus: PreplannedMapArea.PackagingStatus? { get }
     var title: String { get }
     var description: String { get }
     var thumbnail: LoadableImage? { get }
+    var id: PortalItem.ID? { get }
```

Comment:
Shouldn't all preplanned map areas come from a portal item with a valid id? if so this should be non-nullable.